### DISCUSSION TITLE
[Bug Report] 服务工厂多实例初始化 Bug.

### DISCUSSION THREAD

**Discussion by PayFv** [midwayjs/midway#4234] [Q&A]
### Describe the problem（描述问题）

文档中，初始化 `async init()` 是一个异步函数，但是在需要使用多实例的情况下，如果使用 `async init()` 作为初始化函数，会出现错误
`MidwayUseWrongMethodError: 78b4cfc8f0c36c17a30736320d85679a not valid by context.get, Use context.getAsync instead!` ，通过 Debug 发现，该错误是由于将`init()`标记成 async 导致，midwayjs 代码片段：
https://github.com/midwayjs/midway/blob/a7ffc3590bf7e7ec2a393962c152a6a02de78374/packages/core/src/definitions/objectCreator.ts#L90-L100

临时解决方案：
将 `async init()` 修改为 `init()`，去掉 async 声明，但在第一次初始化服务工厂时，由于 `initClients(..)`也时一个 async 函数，所以会先有一个未找到服务的错误。

```
    @Init()
    init() {
        this.initClients(this.config );   // 这是一个 async function。
    }

```

### Midway Versions（Midway 版本）

"@midwayjs/core": "^3.12.0",

**Comment by czy88840616** [comment]
是因为入口用了 get 吧，应该用 getAsync，如无特殊情况，都应该是 getAsync

**Reply by PayFv** [reply]
我看其他的 issue 里将，factory 需要手动 init ，请问这个入口是指在组件 onReady 时候手动初始化吗？

**Reply by PayFv** [reply]
https://www.midwayjs.org/docs/service_factory#%E8%8E%B7%E5%8F%96%E5%AE%9E%E4%BE%8B

这里可以帮忙补一个初始化的最佳实践的案例吗？

**Reply by czy88840616** [reply]
> 我看其他的 issue 里将，factory 需要手动 init ，请问这个入口是指在组件 onReady 时候手动初始化吗？

是的，在 onReady 中 getAsync 一下即可。

**Reply by czy88840616** [reply]
<img width="1283" alt="image" src="https://github.com/user-attachments/assets/83eb6aa6-afc3-467c-aa9e-7556004725a6" />
已经更新
